File: implementation/scipamato/scipamato-common-wicket/src/main/java/ch/difty/scipamato/common/web/model/InjectedLoadableDetachableModel.java
package ch.difty.scipamato.common.web.model;

import java.util.List;

import org.apache.wicket.injection.Injector;
import org.apache.wicket.model.LoadableDetachableModel;

public abstract class InjectedLoadableDetachableModel<T> extends LoadableDetachableModel<List<T>> {
    private static final long serialVersionUID = 1L;

    public InjectedLoadableDetachableModel() {
        injectThis();
    }

    /**
     * protected for overriding to get wicket-free test stubbing
     */
    protected void injectThis() {
        Injector.get()
            .inject(this);
    }
}

File: common/common-wicket/src/main/kotlin/ch/difty/scipamato/common/web/model/InjectedLoadableDetachableModel.kt
package ch.difty.scipamato.common.web.model

import org.apache.wicket.injection.Injector
import org.apache.wicket.model.LoadableDetachableModel

abstract class InjectedLoadableDetachableModel<T> : LoadableDetachableModel<List<T>>() {

    init {
        @Suppress("LeakingThis")
        injectThis()
    }

    /** for overriding to get wicket-free test stubbing */
    protected open fun injectThis() {
        Injector.get().inject(this)
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
